src/components/navigation-header/components/dropdown-menu/index.tsx

```diff
@@ -204,105 +205,120 @@ const NavigationHeaderDropdownMenu = ({
 				id={menuId}
 				style={{ display: isOpen ? 'block' : 'none' }}
 			>
-				{itemGroups.map(
-					(itemGroup: NavigationHeaderItemGroup, groupIndex: number) => {
-						const { items, label } = itemGroup
-						const groupId = generateItemGroupId(groupIndex)
-						const isLastItemGroup = groupIndex === numberOfItemGroups - 1
-						const showDivider = numberOfItemGroups > 1 && !isLastItemGroup
-						const hasLabel = !!label
-						const itemGroupLabelId = hasLabel
-							? `${groupId}-itemGroupLabel`
-							: undefined
-						return (
-							<Fragment key={groupId}>
-								{hasLabel && (
-									<Text
-										asElement="p"
-										className={s.itemGroupLabel}
-										id={itemGroupLabelId}
-										size={100}
-										weight="semibold"
-									>
-										{label}
-									</Text>
-								)}
-								<ul
-									aria-labelledby={itemGroupLabelId}
-									className={classNames(s.itemGroup, {
-										[s.twoColumns]: items.length >= 10,
-									})}
-								>
-									{items.map(
```

Comment:
Nit: Similar to my comment above, perhaps readability could be improved by making `Item` a separate component.

Reply:
@mikegolus This is a good point, and it would definitely help readability. 

Unfortunately, the JSX for `Item` and `itemGroup` rely on many functions defined in `DropdownMenu` and invoked in the JSX. To migrate `Item` and `ItemGroup` to their own components would require significant refactoring to ensure all the functions called by the original code all run as expected, passed in as props or otherwise reimplemented to work 😭 

I will make a note of this and your other comment though, to follow-up on!